components/brave_ads/renderer/brave_ads_render_frame_observer.cc

```diff
@@ -19,17 +21,32 @@ namespace brave_ads {
 
 namespace {
 
-constexpr auto kVettedHosts = base::MakeFixedFlatSet<base::StringPiece>(
-    {"talk.brave.com", "beta.talk.brave.com", "talk.bravesoftware.com",
-     "beta.talk.bravesoftware.com", "talk.brave.software",
-     "beta.talk.brave.software", "dev.talk.brave.software"});
+constexpr auto kVettedBraveTalkHosts =
```

Comment:
i think these vetted hosts should be switched to origins instead of hostnames, because we should not be giving these privileges to them if they are loaded over HTTP instead of HTTPS.

Reply:
thanks, I will do